### DISCUSSION TITLE
gofumpt: rule in v0.3.0 deleted, formatting now conflicts with golangci linting

### DISCUSSION THREAD

**Discussion by katexochen** [golangci/golangci-lint#2638] [Q&A] [answered]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

In https://github.com/mvdan/gofumpt/commit/633346921928018a2cb46ca77800e96d2ab40cf6, a gofumpt rule was deleted. This change is published within the newest release v0.3.0 of gofumpt. The latest release of golangci-lint still uses gofumpt in version v0.2.1, which leads to a conflict. If I format my code with recent version of gofumpt, gloangci-lint will throw an error when running the gofumpt linter. The used version of gofumpt was already bumped in https://github.com/golangci/golangci-lint/commit/5ffadacb04a4aa265f191dfed159c5dc4f307b32, it would be really nice to see a release with this change soon.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.44.2 built from d58dbde5 on 2022-02-17T20:58:06Z
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
cat: .golangci.yml: No such file or directory
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.17.1 linux/amd64
```

</details>

### Verbose output of running

<details>

```console
$ gofumpt -l .
$ golangci-lint run ./... --disable-all --enable gofumpt
xxx.go gofumpt  File is not `gofumpt`-ed
```

</details>

### Code example or link to a public repository

<details>

```go
type foo interface {
	a() error

	b() error
}
```

</details>

**Comment by boring-cyborg** [comment]
Hey, thank you for opening your first Issue ! 🙂 If you would like to contribute we have a [guide for contributors](https://golangci-lint.run/contributing/quick-start/).

**Comment by bombsimon** [comment, marked as answer]
As you already noticed yourself this is not a bug and the version is bumped and will be a part of the next release. This will happen for all linters if they're run separate from `golangci-lint` and the version mismatch. If you're not able to wait for a new release of `golangci-lint`, building a new version from main might be an alternative.

**Reply by serbaut** [reply]
I built from master but still get a diff vs gofumpt v0.3.0 which is very strange.

```
$ go version -m golangci-lint | grep gofumpt
        dep     mvdan.cc/gofumpt        v0.3.0  h1:kTojdZo9AcEYbQYhGuLf/zszYthRdhDNDUi2JKTxas4=
$ gofumpt -version
v0.3.0
$ gofumpt -l .
$ ./golangci-lint run --enable=gofumpt ./...
foo/bar.go:8: File is not `gofumpt`-ed (gofumpt)
```
